test/test-cmake.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# Copyright (c) 2023 Huang Qinjin
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+. "${0%/*}/test.sh"
+
+
+CMAKE_ARGS=(
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    -DCMAKE_SYSTEM_NAME=Windows
+)
+
+case $OSTYPE in
+    darwin*)
+        CMAKE_ARGS+=(
+            # mt.exe always crashes on macOS.
+            -DCMAKE_EXE_LINKER_FLAGS=/MANIFEST:NO
+            -DCMAKE_SHARED_LINKER_FLAGS=/MANIFEST:NO
+            # No winbind package available on macOS.
+            # https://github.com/mstorsjo/msvc-wine/issues/6
+            -DCMAKE_MSVC_DEBUG_INFORMATION_FORMAT=Embedded
+        ) ;;
+esac
+
+EXEC "" CC=${BIN}cl CXX=${BIN}cl RC=${BIN}rc cmake -S"$TESTS" -GNinja "${CMAKE_ARGS[@]}"
+EXEC "" ninja -v
+
+# Rerun ninja to make sure that dependencies aren't broken.
+EXEC ninja-rerun ninja -d explain -v
+DIFF ninja-rerun.out - <<EOF
+ninja: no work to do.
```

Comment:
If this testcase fails, the explanation is stored in `ninja-rerun.err`, but that's not printed anywhere. It'd be nice if that was printed, especially if the diff test failed.

A second note; the diff is a bit hard to read when it's reversed (the actual output is the original state and the expected output is the new state, in the diff); it's easy to change by swapping the sides in the diff into `DIFF - ninja-rerun.out`, but that makes it slightly harder to read when the output name doesn't line up with the `EXEC` line above.